UnitTests/GitCommandsTests/Git/Gpg/GitGpgControllerTests.cs

```diff
@@ -41,63 +41,27 @@ public void Validate_GetRevisionCommitSignatureStatusAsync(CommitStatus expected
         {
             var guid = Guid.NewGuid().ToString("N");
 
-            GitRevision revision = new GitRevision(null, "");
+            GitRevision revision = new GitRevision(null, guid);
 
+            _module().RunGitCmd($"log --pretty=\"format:%G?\" -1 {revision.Guid}").Returns(gitCmdReturn);
+            
             var actual = await _gpgController.GetRevisionCommitSignatureStatusAsync(revision);
 
-            _module().RunGitCmd($"log --pretty=\"format:%G?\" -1 {guid}").Returns(gitCmdReturn);
-
             Assert.AreEqual(expected, actual);
         }
 
 
         [TestCase]
         public void Validate_GetRevisionCommitSignatureStatusAsync_null_revision()
         {
-            ExecuteGetRevisionCommitSignatureStatusAsync_null(null);
-        }
-
-        private async void ExecuteGetRevisionCommitSignatureStatusAsync_null(GitRevision revision)
-        {
-            try
-            {
-                await _gpgController.GetRevisionCommitSignatureStatusAsync(revision);
-            }
-            catch (ArgumentNullException)
-            {
-                // assertion passed if exception thrown
-                Assert.AreEqual(true, true);
-                return;
-            }
-
-            // assertion not passed if exception NOT thrown
-            Assert.AreEqual(true, false);
+            ((Func<Task>)(async () => await _gpgController.GetRevisionCommitSignatureStatusAsync(null))).ShouldThrow<ArgumentNullException>();
         }
-
-
-
+        
 
         [TestCase]
         public void Validate_GetRevisionTagSignatureStatusAsync_null_revision()
         {
-            ExecuteGetRevisionTagSignatureStatusAsync_null(null);
-        }
-
-        private async void ExecuteGetRevisionTagSignatureStatusAsync_null(GitRevision revision)
-        {
-            try
-            {
-                await _gpgController.GetRevisionTagSignatureStatusAsync(revision);
-            }
-            catch (ArgumentNullException)
-            {
-                // assertion passed if exception thrown
-                Assert.AreEqual(true, true);
-                return;
-            }
-
-            // assertion not passed if exception NOT thrown
-            Assert.AreEqual(true, false);
+            ((Func<Task>)(async () => await _gpgController.GetRevisionTagSignatureStatusAsync(null))).ShouldThrow<ArgumentNullException>();
```

Comment:
what about the remaining unit tests?
most (if not all) of them can be simplified.

Reply:
Now I'm not on win and I'm having problem with github. 
Next days I'll try to rebase again to avoid conflict on GitCommands.csproj and changing some tests to see if they works.